tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+    def consume(self, wait=True):
+        """Starts consume messages thread
+
+        Args:
+            wait (bool, optional): Wait for consume thread to finish or not.
+            Defaults to True.
+        """
+        self.logger.info("Start of consuming messages")
+        thread = threading.Thread(name="stream_consume_thread",
+                                  target=self._worker_thread,
+                                  args=(self._consume_from_topic, self.workers,
+                                        self.topics, self.total_messages,
+                                        self.logger))
+        thread.start()
+        self.consume_thread = thread
+        if wait:
+            thread.join()
+        return
+
+    @staticmethod
+    def _consume_from_topic(logger: logging.Logger, topic: TopicStatus):
+        """Consumes all messages from topic
+
+        Args:
+            logger (logging.Logger): logger class
+            topic (TopicStatus): current live topic status class to work with
+
+        Raises:
+            ck.KafkaException: on any error occured while consuming
+
+        Returns:
+            int: number of consumed messages
+        """
+        def time_since_last_msg() -> int:
+            diff_ms = datetime.now().timestamp() - topic.last_message_ts
+            return int(diff_ms * 1000)
+
+        topic.last_message_ts = datetime.now().timestamp()
+
+        # Message consuming loop
+        consumer = ck.Consumer(topic.consumer_config)
+        try:
+            # Recent changes offers use of subscribe
+            # regardless of transactions and partitions
+            consumer.subscribe([topic.name])
+            while True:
+                # calculate elapsed time
+                _since_last_msg_ms = time_since_last_msg()
+
+                # Exit on timeout
+                if _since_last_msg_ms > topic.consume_timeout:
+                    logger.error("Timeout consuming messages "
+                                 f"from {topic.name}")
+                    break
+
+                # Poll for the message
+                msg = consumer.poll(timeout=1.0)
+                if msg is None:
+                    # no messages
+                    continue
+                # On error, check for the EOF
+                if msg.error():
+                    if msg.error().code() == ck.KafkaError._PARTITION_EOF:
+                        # End of partition event
+                        logger.info(f"Consumer of '{msg.topic()}' "
+                                    f"[{msg.partition()}] reached "
+                                    f"end at offset {msg.offset()}")
+                        break
+                    # If not EOF, raise the error
+                    elif msg.error():
+                        raise ck.KafkaException(msg.error())
+                else:
+                    topic.last_message_ts = datetime.now().timestamp()
+                    # Save index
+                    topic.index += 1
```

Comment:
Its a bit unclear what exactly this is doing? Is this supposed to "verify" something?

Reply:
This is just a number of consumed messages. As I get from different comment, main atomic consume produce routing is missed here